libs/geospatial/src/main/java/org/codice/ddf/libs/geo/coordinate/Angle.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.libs.geo.coordinate;
+
+/**
+ * Abstract class that is the base for Latitude and Longitude coordinate values
+ */
+abstract class Angle {
+
+    protected double decimalDegrees;
+
+    protected int degree;
+
+    protected int minute;
+
+    protected int second;
+
+    protected int centisecond;
+
+    /**
+     * Constructor which sets value to 0
+     */
+    public Angle() {
+        this(0);
+    }
+
+    /**
+     * Constructor for Decimal Degree Coordinate values
+     *
+     * @param decimalDegrees the Decimal Degree Coordinate value
+     */
+    public Angle(double decimalDegrees) {
+        calculateDMS(decimalDegrees);
+        this.decimalDegrees = decimalDegrees;
+    }
+
+    /**
+     * Calculates the DMS value for the Decimal Degree Coordinate value
+     *
+     * @param decimalDegrees the Decimal Degree value
+     */
+    protected void calculateDMS(double decimalDegrees) {
+        degree = Math.abs((int) decimalDegrees);
+
+        minute = (int) ((Math.abs(decimalDegrees) - degree) * 60);
+
+        second = (int) ((((Math.abs(decimalDegrees) - degree) * 60) - minute) * 60);
+
+        centisecond = (int) (Math.min(99,
+                Math.round((((((Math.abs(decimalDegrees) - degree) * 60) - minute) * 60) - second)
+                        * 100)));
+    }
+
+    /**
+     * Calculates the Decimal Degree value for the DMS values
+     * <p>
+     * Note: an IllegalArgumentException will be thrown if any of the values are not in the valid ranges
+     *
+     * @param degrees      the DMS Degree value
+     * @param minutes      the DMS Minute value
+     * @param seconds      the DMS Second value
+     * @param centiseconds the DMS Centisecond value
```

Comment:
This probably needs more explanation. Passing a fractional value as an integer as the fraction of a second is really weird API. That is, the legal values are 0 to 999, and its centiseconds, so if I pass 999, its really 9.99 seconds. Can I suggest milliseconds or just allow the seconds to be a floating point value?